Kitodo-API/src/main/java/org/kitodo/api/ugh/DocStructInterface.java

```diff
@@ -0,0 +1,628 @@
+     *
+     * @return all meta-data types that can be added to this instance. A return
+     *         type of {@code Collection<>} would be sufficient.
+     */
+    List<MetadataTypeInterface> getPossibleMetadataTypes();
+
+    /**
+     * Get the type of this DocStruct.
+     *
+     * @return the type of this DocStruct
+     */
+    DocStructTypeInterface getDocStructType();
+
+    /**
+     * Returns whether a {@code DocStruct} of the given {@code DocStructType} is
+     * allowed to be added to this instance.
+     *
+     * @param inType
+     *            the {@code DocStructType} in question
+     * @return true, if {@code DocStruct} of this type can be added; otherwise
+     *         false
+     */
+    boolean isDocStructTypeAllowedAsChild(DocStructTypeInterface type);
+
+    /**
+     * Removes a child from this instance.
+     *
+     * @return whether the list contained that child. The return value is never
+     *         used.
+     */
+    boolean removeChild(DocStructInterface docStructInterface);
+
+    /**
+     * @return whether the content file references list was initialized with an
+     *         list instance before the function call. The return value is never
+     *         used.
+     */
+    boolean removeContentFile(ContentFileInterface contentFileInterface) throws ContentFileNotLinkedException;
+
+    /**
+     * Removes all links from this instance to a given content file. If the
+     * given {@code ContentFile} is referenced more than once from this
+     * instance, all links are removed. For that reason, all attached
+     * {@code ContentFileReference} objects are searched.
+     *
+     * @param theContentFile
+     *            the content file to be removed
+     * @return always {@code true}. The return value is never used.
+     */
+    boolean removeMetadata(MetadataInterface md);
+
+    /**
+     * Removes a meta-data group from this instance. If (according to
+     * configuration) at least one meta-data group of this type is required on
+     * this instance, the meta-data group will <i>not be removed</i>.
+     *
+     * <p>
+     * If you want to remove a meta-data group just to replace it, use the
+     * method {{@code changeMetadataGroup(MetadataGroup, MetadataGroup)}
+     * instead.
+     *
+     * @param metadataGroup
+     *            meta-data group which should be removed
+     * @return always {@code true}. The return value is never used.
+     */
+    boolean removeMetadataGroup(MetadataGroupInterface metadataGroup);
+
+    /**
+     * Removes a person from this instance.
+     *
+     * @param in
+     *            person which should be removed
+     * @return whether the persons list was initialized with an list instance
+     *         before the function call. The return value is never used.
+     * @throws IncompletePersonObjectException
+     *             if {@code in} does not have a {@code MetadataType}
+     */
+    boolean removePerson(PersonInterface p);
```

Comment:
1) See my comment to `setLocation`.
2) Please adjust the JavaDoc or method signature as they are different.